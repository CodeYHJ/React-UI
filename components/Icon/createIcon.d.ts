export interface CreateIconProps {
    scriptUrl: string;
}
declare const createIcon: (scriptUrl: string) => import("react").SFC<import("./localIcon").IconProps>;
export default createIcon;
