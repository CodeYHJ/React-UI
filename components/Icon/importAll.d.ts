declare let importAll: (requireContext: any) => any;
