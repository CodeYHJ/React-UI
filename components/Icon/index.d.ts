import I from './localIcon';
import c from './createIcon';
declare type IconType = typeof I & {
    createIcon: typeof c;
};
declare const Icon: IconType;
export default Icon;
