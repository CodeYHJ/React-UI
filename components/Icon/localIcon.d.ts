import React, { SVGAttributes } from 'react';
import "./importAll";
import "./style/index.less";
export interface IconProps extends SVGAttributes<SVGElement> {
    name: string;
}
declare const Icon: React.SFC<IconProps>;
export default Icon;
