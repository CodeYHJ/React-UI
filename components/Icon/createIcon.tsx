import Icon from './localIcon';

export interface CreateIconProps {
  scriptUrl: string;
}

const setMap = new Set();

const createIcon = (scriptUrl: string) => {
  if (!setMap.has(scriptUrl)) {
    const svgDom = document.createElement('script');

    svgDom.setAttribute('src', scriptUrl);

    document.body.appendChild(svgDom);

    setMap.add(scriptUrl);
  }

  return Icon;
};

export default createIcon;
