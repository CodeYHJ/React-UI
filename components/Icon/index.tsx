import I from './localIcon';

import c from './createIcon';

export * from './svgIcon';

type IconType = typeof I & { createIcon: typeof c };

const Icon = I as IconType;

Icon.createIcon = c;

export default Icon;
