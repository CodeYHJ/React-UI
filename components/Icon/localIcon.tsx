import React, { SVGAttributes } from 'react';

import './style/index.less';

import { classPre } from '@com/utils';

export interface IconProps extends SVGAttributes<SVGElement> {
  name: string;
}

const c = classPre('svg');

const Icon: React.FunctionComponent<IconProps> = (props) => {
  const { className, name, ...others } = props;

  const cls = [c(), className].filter(Boolean).join(' ');

  return (
    <svg className={cls} {...others}>
      <use xlinkHref={`#${name}`} />
    </svg>
  );
};

export default Icon;
