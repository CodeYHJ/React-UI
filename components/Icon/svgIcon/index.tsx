export { default as ArrorDownIcon } from './ArrorDownIcon';

export { default as ArrorLeftIcon } from './ArrorLeftIcon';

export { default as ArrorRightIcon } from './ArrorRightIcon';

export { default as ArrorUpIcon } from './ArrorUpIcon';

export { default as CancelIcon } from './CancelIcon';

export { default as CodeIcon } from './CodeIcon';

export { default as DangerIcon } from './DangerIcon';

export { default as LoadingIcon } from './LoadingIcon';

export { default as MainIcon } from './MainIcon';

export { default as PasswordIcon } from './PasswordIcon';

export { default as SuccessIcon } from './SuccessIcon';

export { default as UserIcon } from './UserIcon';

export { default as WarnIcon } from './WarnIcon';
