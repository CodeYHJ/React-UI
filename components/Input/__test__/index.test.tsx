import React from 'react';
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import { Input } from '@com/index';
import { UserIcon, PasswordIcon } from '@com/Icon';
describe('<Input/>', () => {
    describe("type", () => {
        it('renders an `type=text` component', () => {
            const json = renderer.create(<Input placeholder="text" />)
            expect(json).toMatchSnapshot()
        })
        it('renders an `type=password` component', () => {
            const json = renderer.create(<Input type="password" placeholder="password" />)
            expect(json).toMatchSnapshot()
        })
        it('renders an `type=number` component', () => {
            const json = renderer.create(<Input type="number" placeholder="number" />)
            expect(json).toMatchSnapshot()
        })
    })
    describe('event', () => {
        it('should handle onChange Event', () => {
            const fn = jest.fn();
            const wrapper = shallow(<Input onChange={fn} />);
            wrapper.find('.code-ui-input-inputLabel').simulate('change', {
                target: {
                    value: 'change'
                }
            })
            expect(fn).toHaveBeenCalled();
            expect(fn).toBeCalledWith({
                target: {
                    value: 'change'
                }
            });
        })
        it('should handle onFocus Event', () => {
            const fn = jest.fn();
            const wrapper = shallow(<Input onFocus={fn} />);
            wrapper.find('.code-ui-input-inputLabel').simulate('focus')
            expect(fn).toHaveBeenCalled();
        })
        it('should handle onBlur Event', () => {
            const fn = jest.fn();
            const wrapper = shallow(<Input onBlur={fn} />);
            wrapper.find('.code-ui-input-inputLabel').simulate('blur')
            expect(fn).toHaveBeenCalled();
        })
    })
    describe('iconFix', () => {
        const userIcon = <UserIcon />
        const passwordIcon = <PasswordIcon />
        it('renders an `preIconFix` component ', () => {
            const json = renderer.create(<Input placeholder="前置Icon" preIconFix={userIcon} />)
            expect(json).toMatchSnapshot()
        })
        it('renders an `sufIconFix` component ', () => {
            const json = renderer.create(<Input placeholder="后置Icon" sufIconFix={passwordIcon} />)
            expect(json).toMatchSnapshot()
        })
        it('renders an `preIconFix and sufIconFix` component ', () => {
            const json = renderer.create(<Input placeholder="前/后置Icon" preIconFix={userIcon} sufIconFix={passwordIcon} />)
            expect(json).toMatchSnapshot()
        })
    })
    describe('strFix', () => {
        it('renders an `preStrFix` component ', () => {
            const json = renderer.create(<Input placeholder="baidu.com" preStrFix='http://' />)
            expect(json).toMatchSnapshot()
        })
        it('renders an `sufStrFix` component ', () => {
            const json = renderer.create(<Input placeholder="http://baidu" sufStrFix='.com' />)
            expect(json).toMatchSnapshot()
        })
        it('renders an `preStrFix and sufStrFix` component ', () => {
            const json = renderer.create(<Input placeholder="baidu" preStrFix='http://' sufStrFix='.com' />)
            expect(json).toMatchSnapshot()
        })
    })
    describe('status', () => {
        const userIcon = <UserIcon />
        const passwordIcon = <PasswordIcon />
        describe('render', () => {
            describe('strFix Error', () => {
                it('renders an `preStrFix Error` component ', () => {
                    const json = renderer.create(<Input placeholder="固定文字块 Error" preStrFix='http://' error />)
                    expect(json).toMatchSnapshot()
                })
                it('renders an `sufStrFix Error` component ', () => {
                    const json = renderer.create(<Input placeholder="http://baidu" sufStrFix='.com' error />)
                    expect(json).toMatchSnapshot()
                })
                it('renders an `preStrFix and sufStrFix Error` component ', () => {
                    const json = renderer.create(<Input placeholder="baidu" preStrFix='http://' sufStrFix='.com' error />)
                    expect(json).toMatchSnapshot()
                })
            })
            describe('strFix Disabled', () => {
                it('renders an `preStrFix Disabled` component ', () => {
                    const json = renderer.create(<Input placeholder="固定文字块 Error" preStrFix='http://' disabled />)
                    expect(json).toMatchSnapshot()
                })
                it('renders an `sufStrFix Disabled` component ', () => {
                    const json = renderer.create(<Input placeholder="http://baidu" sufStrFix='.com' disabled />)
                    expect(json).toMatchSnapshot()
                })
                it('renders an `preStrFix and sufStrFix Disabled` component ', () => {
                    const json = renderer.create(<Input placeholder="baidu" preStrFix='http://' sufStrFix='.com' disabled />)
                    expect(json).toMatchSnapshot()
                })
            })
            describe('iconFix Error', () => {
                it('renders an `preIconFix Error` component ', () => {
                    const json = renderer.create(<Input placeholder="前置Icon" preIconFix={userIcon} error />)
                    expect(json).toMatchSnapshot()
                })
                it('renders an `sufIconFix Error` component ', () => {
                    const json = renderer.create(<Input placeholder="后置Icon" sufIconFix={passwordIcon} error />)
                    expect(json).toMatchSnapshot()
                })
                it('renders an `preIconFix and sufIconFix Error` component ', () => {
                    const json = renderer.create(<Input placeholder="前/后置Icon" preIconFix={userIcon} sufIconFix={passwordIcon} error />)
                    expect(json).toMatchSnapshot()
                })
            })
            describe('iconFix Disabled', () => {
                it('renders an `preIconFix Disabled` component ', () => {
                    const json = renderer.create(<Input placeholder="前置Icon" preIconFix={userIcon} disabled />)
                    expect(json).toMatchSnapshot()
                })
                it('renders an `sufIconFix Disabled` component ', () => {
                    const json = renderer.create(<Input placeholder="后置Icon" sufIconFix={passwordIcon} disabled />)
                    expect(json).toMatchSnapshot()
                })
                it('renders an `preIconFix and sufIconFix Disabled` component ', () => {
                    const json = renderer.create(<Input placeholder="前/后置Icon" preIconFix={userIcon} sufIconFix={passwordIcon} disabled />)
                    expect(json).toMatchSnapshot()
                })
            })
        })

    })
})