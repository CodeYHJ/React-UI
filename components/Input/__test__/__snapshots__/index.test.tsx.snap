// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> iconFix renders an \`preIconFix and sufIconFix\` component  1`] = `
<span
  className="code-ui-input-allIconFix"
  onClick={[Function]}
>
  <span
    className="code-ui-input-icon"
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M512 160c-123.36 0-224 100.64-224 224 0 77.12 39.36 145.6 99.008 186.016C272.864 618.976 192 732.256 192 864h64c0-141.76 114.24-256 256-256s256 114.24 256 256h64c0-131.744-80.864-244.992-195.008-294.016A224.768 224.768 0 00736 384c0-123.36-100.64-224-224-224zm0 64c88.736 0 160 71.264 160 160s-71.264 160-160 160-160-71.264-160-160 71.264-160 160-160z"
      />
    </svg>
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="前/后置Icon"
    type="text"
  />
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M785.7 360.1H724v-75.4c0-58.4-22.7-113.3-64-154.5-41.3-41.3-96.2-64-154.5-64s-113.3 22.7-154.5 64c-41.3 41.3-64 96.2-64 154.5v75.4h-61.8c-56.1 0-101.8 45.7-101.8 101.8v396.6c0 56.1 45.7 101.8 101.8 101.8h560.7c56.1 0 101.8-45.7 101.8-101.8V461.9c-.1-56.2-45.8-101.8-102-101.8zm-434.8-75.4c0-85.2 69.3-154.6 154.6-154.6S660 199.5 660 284.7v75.4H350.9v-75.4zm472.7 573.8c0 20.9-17 37.8-37.8 37.8H225.1c-20.9 0-37.8-17-37.8-37.8V461.9c0-20.9 17-37.8 37.8-37.8h560.7c20.9 0 37.8 17 37.8 37.8v396.6z"
      />
      <path
        d="M505.4 566.3c-17.7 0-32 14.3-32 32v123.8c0 17.7 14.3 32 32 32s32-14.3 32-32V598.3c0-17.7-14.3-32-32-32z"
      />
    </svg>
  </span>
</span>
`;

exports[`<Input/> iconFix renders an \`preIconFix\` component  1`] = `
<span
  className="code-ui-input-preIconFix"
>
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M512 160c-123.36 0-224 100.64-224 224 0 77.12 39.36 145.6 99.008 186.016C272.864 618.976 192 732.256 192 864h64c0-141.76 114.24-256 256-256s256 114.24 256 256h64c0-131.744-80.864-244.992-195.008-294.016A224.768 224.768 0 00736 384c0-123.36-100.64-224-224-224zm0 64c88.736 0 160 71.264 160 160s-71.264 160-160 160-160-71.264-160-160 71.264-160 160-160z"
      />
    </svg>
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="前置Icon"
    type="text"
  />
</span>
`;

exports[`<Input/> iconFix renders an \`sufIconFix\` component  1`] = `
<span
  className="code-ui-input-sufIconFix"
>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="后置Icon"
    type="text"
  />
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M785.7 360.1H724v-75.4c0-58.4-22.7-113.3-64-154.5-41.3-41.3-96.2-64-154.5-64s-113.3 22.7-154.5 64c-41.3 41.3-64 96.2-64 154.5v75.4h-61.8c-56.1 0-101.8 45.7-101.8 101.8v396.6c0 56.1 45.7 101.8 101.8 101.8h560.7c56.1 0 101.8-45.7 101.8-101.8V461.9c-.1-56.2-45.8-101.8-102-101.8zm-434.8-75.4c0-85.2 69.3-154.6 154.6-154.6S660 199.5 660 284.7v75.4H350.9v-75.4zm472.7 573.8c0 20.9-17 37.8-37.8 37.8H225.1c-20.9 0-37.8-17-37.8-37.8V461.9c0-20.9 17-37.8 37.8-37.8h560.7c20.9 0 37.8 17 37.8 37.8v396.6z"
      />
      <path
        d="M505.4 566.3c-17.7 0-32 14.3-32 32v123.8c0 17.7 14.3 32 32 32s32-14.3 32-32V598.3c0-17.7-14.3-32-32-32z"
      />
    </svg>
  </span>
</span>
`;

exports[`<Input/> status render iconFix Disabled renders an \`preIconFix Disabled\` component  1`] = `
<span
  className="code-ui-input-preIconFix code-ui-input-iconDisabled"
>
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M512 160c-123.36 0-224 100.64-224 224 0 77.12 39.36 145.6 99.008 186.016C272.864 618.976 192 732.256 192 864h64c0-141.76 114.24-256 256-256s256 114.24 256 256h64c0-131.744-80.864-244.992-195.008-294.016A224.768 224.768 0 00736 384c0-123.36-100.64-224-224-224zm0 64c88.736 0 160 71.264 160 160s-71.264 160-160 160-160-71.264-160-160 71.264-160 160-160z"
      />
    </svg>
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="前置Icon"
    type="text"
  />
</span>
`;

exports[`<Input/> status render iconFix Disabled renders an \`preIconFix and sufIconFix Disabled\` component  1`] = `
<span
  className="code-ui-input-allIconFix code-ui-input-iconDisabled"
  onClick={[Function]}
>
  <span
    className="code-ui-input-icon"
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M512 160c-123.36 0-224 100.64-224 224 0 77.12 39.36 145.6 99.008 186.016C272.864 618.976 192 732.256 192 864h64c0-141.76 114.24-256 256-256s256 114.24 256 256h64c0-131.744-80.864-244.992-195.008-294.016A224.768 224.768 0 00736 384c0-123.36-100.64-224-224-224zm0 64c88.736 0 160 71.264 160 160s-71.264 160-160 160-160-71.264-160-160 71.264-160 160-160z"
      />
    </svg>
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="前/后置Icon"
    type="text"
  />
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M785.7 360.1H724v-75.4c0-58.4-22.7-113.3-64-154.5-41.3-41.3-96.2-64-154.5-64s-113.3 22.7-154.5 64c-41.3 41.3-64 96.2-64 154.5v75.4h-61.8c-56.1 0-101.8 45.7-101.8 101.8v396.6c0 56.1 45.7 101.8 101.8 101.8h560.7c56.1 0 101.8-45.7 101.8-101.8V461.9c-.1-56.2-45.8-101.8-102-101.8zm-434.8-75.4c0-85.2 69.3-154.6 154.6-154.6S660 199.5 660 284.7v75.4H350.9v-75.4zm472.7 573.8c0 20.9-17 37.8-37.8 37.8H225.1c-20.9 0-37.8-17-37.8-37.8V461.9c0-20.9 17-37.8 37.8-37.8h560.7c20.9 0 37.8 17 37.8 37.8v396.6z"
      />
      <path
        d="M505.4 566.3c-17.7 0-32 14.3-32 32v123.8c0 17.7 14.3 32 32 32s32-14.3 32-32V598.3c0-17.7-14.3-32-32-32z"
      />
    </svg>
  </span>
</span>
`;

exports[`<Input/> status render iconFix Disabled renders an \`sufIconFix Disabled\` component  1`] = `
<span
  className="code-ui-input-sufIconFix code-ui-input-iconDisabled"
>
  <input
    className="code-ui-input-inputLabel"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="后置Icon"
    type="text"
  />
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M785.7 360.1H724v-75.4c0-58.4-22.7-113.3-64-154.5-41.3-41.3-96.2-64-154.5-64s-113.3 22.7-154.5 64c-41.3 41.3-64 96.2-64 154.5v75.4h-61.8c-56.1 0-101.8 45.7-101.8 101.8v396.6c0 56.1 45.7 101.8 101.8 101.8h560.7c56.1 0 101.8-45.7 101.8-101.8V461.9c-.1-56.2-45.8-101.8-102-101.8zm-434.8-75.4c0-85.2 69.3-154.6 154.6-154.6S660 199.5 660 284.7v75.4H350.9v-75.4zm472.7 573.8c0 20.9-17 37.8-37.8 37.8H225.1c-20.9 0-37.8-17-37.8-37.8V461.9c0-20.9 17-37.8 37.8-37.8h560.7c20.9 0 37.8 17 37.8 37.8v396.6z"
      />
      <path
        d="M505.4 566.3c-17.7 0-32 14.3-32 32v123.8c0 17.7 14.3 32 32 32s32-14.3 32-32V598.3c0-17.7-14.3-32-32-32z"
      />
    </svg>
  </span>
</span>
`;

exports[`<Input/> status render iconFix Error renders an \`preIconFix Error\` component  1`] = `
<span
  className="code-ui-input-preIconFix code-ui-input-iconError"
>
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M512 160c-123.36 0-224 100.64-224 224 0 77.12 39.36 145.6 99.008 186.016C272.864 618.976 192 732.256 192 864h64c0-141.76 114.24-256 256-256s256 114.24 256 256h64c0-131.744-80.864-244.992-195.008-294.016A224.768 224.768 0 00736 384c0-123.36-100.64-224-224-224zm0 64c88.736 0 160 71.264 160 160s-71.264 160-160 160-160-71.264-160-160 71.264-160 160-160z"
      />
    </svg>
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="前置Icon"
    type="text"
  />
</span>
`;

exports[`<Input/> status render iconFix Error renders an \`preIconFix and sufIconFix Error\` component  1`] = `
<span
  className="code-ui-input-allIconFix code-ui-input-iconError"
  onClick={[Function]}
>
  <span
    className="code-ui-input-icon"
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M512 160c-123.36 0-224 100.64-224 224 0 77.12 39.36 145.6 99.008 186.016C272.864 618.976 192 732.256 192 864h64c0-141.76 114.24-256 256-256s256 114.24 256 256h64c0-131.744-80.864-244.992-195.008-294.016A224.768 224.768 0 00736 384c0-123.36-100.64-224-224-224zm0 64c88.736 0 160 71.264 160 160s-71.264 160-160 160-160-71.264-160-160 71.264-160 160-160z"
      />
    </svg>
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="前/后置Icon"
    type="text"
  />
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M785.7 360.1H724v-75.4c0-58.4-22.7-113.3-64-154.5-41.3-41.3-96.2-64-154.5-64s-113.3 22.7-154.5 64c-41.3 41.3-64 96.2-64 154.5v75.4h-61.8c-56.1 0-101.8 45.7-101.8 101.8v396.6c0 56.1 45.7 101.8 101.8 101.8h560.7c56.1 0 101.8-45.7 101.8-101.8V461.9c-.1-56.2-45.8-101.8-102-101.8zm-434.8-75.4c0-85.2 69.3-154.6 154.6-154.6S660 199.5 660 284.7v75.4H350.9v-75.4zm472.7 573.8c0 20.9-17 37.8-37.8 37.8H225.1c-20.9 0-37.8-17-37.8-37.8V461.9c0-20.9 17-37.8 37.8-37.8h560.7c20.9 0 37.8 17 37.8 37.8v396.6z"
      />
      <path
        d="M505.4 566.3c-17.7 0-32 14.3-32 32v123.8c0 17.7 14.3 32 32 32s32-14.3 32-32V598.3c0-17.7-14.3-32-32-32z"
      />
    </svg>
  </span>
</span>
`;

exports[`<Input/> status render iconFix Error renders an \`sufIconFix Error\` component  1`] = `
<span
  className="code-ui-input-sufIconFix code-ui-input-iconError"
>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="后置Icon"
    type="text"
  />
  <span
    className="code-ui-input-icon"
    onClick={[Function]}
  >
    <svg
      className="code-ui-svg"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M785.7 360.1H724v-75.4c0-58.4-22.7-113.3-64-154.5-41.3-41.3-96.2-64-154.5-64s-113.3 22.7-154.5 64c-41.3 41.3-64 96.2-64 154.5v75.4h-61.8c-56.1 0-101.8 45.7-101.8 101.8v396.6c0 56.1 45.7 101.8 101.8 101.8h560.7c56.1 0 101.8-45.7 101.8-101.8V461.9c-.1-56.2-45.8-101.8-102-101.8zm-434.8-75.4c0-85.2 69.3-154.6 154.6-154.6S660 199.5 660 284.7v75.4H350.9v-75.4zm472.7 573.8c0 20.9-17 37.8-37.8 37.8H225.1c-20.9 0-37.8-17-37.8-37.8V461.9c0-20.9 17-37.8 37.8-37.8h560.7c20.9 0 37.8 17 37.8 37.8v396.6z"
      />
      <path
        d="M505.4 566.3c-17.7 0-32 14.3-32 32v123.8c0 17.7 14.3 32 32 32s32-14.3 32-32V598.3c0-17.7-14.3-32-32-32z"
      />
    </svg>
  </span>
</span>
`;

exports[`<Input/> status render strFix Disabled renders an \`preStrFix Disabled\` component  1`] = `
<span
  className="code-ui-input-preStrFix code-ui-input-strDisabled"
>
  <div
    className="code-ui-input-str"
    onClick={[Function]}
  >
    http://
  </div>
  <input
    className="code-ui-input-inputLabel"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="固定文字块 Error"
    type="text"
  />
</span>
`;

exports[`<Input/> status render strFix Disabled renders an \`preStrFix and sufStrFix Disabled\` component  1`] = `
<span
  className="code-ui-input-allStrFix code-ui-input-strDisabled"
  onClick={[Function]}
>
  <span
    className="code-ui-input-str"
  >
    http://
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="baidu"
    type="text"
  />
  <span
    className="code-ui-input-str"
    onClick={[Function]}
  >
    .com
  </span>
</span>
`;

exports[`<Input/> status render strFix Disabled renders an \`sufStrFix Disabled\` component  1`] = `
<span
  className="code-ui-input-sufStrFix code-ui-input-strDisabled"
>
  <input
    className="code-ui-input-inputLabel"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="http://baidu"
    type="text"
  />
  <div
    className="code-ui-input-str"
    onClick={[Function]}
  >
    .com
  </div>
</span>
`;

exports[`<Input/> status render strFix Error renders an \`preStrFix Error\` component  1`] = `
<span
  className="code-ui-input-preStrFix code-ui-input-strError"
>
  <div
    className="code-ui-input-str"
    onClick={[Function]}
  >
    http://
  </div>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="固定文字块 Error"
    type="text"
  />
</span>
`;

exports[`<Input/> status render strFix Error renders an \`preStrFix and sufStrFix Error\` component  1`] = `
<span
  className="code-ui-input-allStrFix code-ui-input-strError"
  onClick={[Function]}
>
  <span
    className="code-ui-input-str"
  >
    http://
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="baidu"
    type="text"
  />
  <span
    className="code-ui-input-str"
    onClick={[Function]}
  >
    .com
  </span>
</span>
`;

exports[`<Input/> status render strFix Error renders an \`sufStrFix Error\` component  1`] = `
<span
  className="code-ui-input-sufStrFix code-ui-input-strError"
>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="http://baidu"
    type="text"
  />
  <div
    className="code-ui-input-str"
    onClick={[Function]}
  >
    .com
  </div>
</span>
`;

exports[`<Input/> strFix renders an \`preStrFix and sufStrFix\` component  1`] = `
<span
  className="code-ui-input-allStrFix"
  onClick={[Function]}
>
  <span
    className="code-ui-input-str"
  >
    http://
  </span>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="baidu"
    type="text"
  />
  <span
    className="code-ui-input-str"
    onClick={[Function]}
  >
    .com
  </span>
</span>
`;

exports[`<Input/> strFix renders an \`preStrFix\` component  1`] = `
<span
  className="code-ui-input-preStrFix"
>
  <div
    className="code-ui-input-str"
    onClick={[Function]}
  >
    http://
  </div>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="baidu.com"
    type="text"
  />
</span>
`;

exports[`<Input/> strFix renders an \`sufStrFix\` component  1`] = `
<span
  className="code-ui-input-sufStrFix"
>
  <input
    className="code-ui-input-inputLabel"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="http://baidu"
    type="text"
  />
  <div
    className="code-ui-input-str"
    onClick={[Function]}
  >
    .com
  </div>
</span>
`;

exports[`<Input/> type renders an \`type=number\` component 1`] = `
<input
  className="code-ui-input-inputLabel"
  disabled={false}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  placeholder="number"
  type="number"
/>
`;

exports[`<Input/> type renders an \`type=password\` component 1`] = `
<input
  className="code-ui-input-inputLabel"
  disabled={false}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  placeholder="password"
  type="password"
/>
`;

exports[`<Input/> type renders an \`type=text\` component 1`] = `
<input
  className="code-ui-input-inputLabel"
  disabled={false}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  placeholder="text"
  type="text"
/>
`;
